### PR TITLE
[Sophia]Tasks 41 to 50

### PR DESCRIPTION
**Pull request by SofiMat**

### CODE REVIEW

File: tasks/42_first-n-smallest/first-n-smallest.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        arr[j] = arr[j + 1];
+        j++;
+      }
+      arr[k - 1] = arr[i];
+    }
+  }
+  for (let i = 0; i < k; i++) {
+    final.push(arr[i]);
+  }
+  return final;
```
Comment (prezlyataa) [inline tasks/42_first-n-smallest/first-n-smallest.js]: I this case if you pass such arguments `firstNSmallest([1, 2, 3, 1, 2], 10)` your result must be origin array with length that equal 5, but you get result with length that equal 10 with five properly values and five empty. You should just cut your result, for example `return final.slice(0,arr.length);`

File: tasks/45_equal-sides/equal-sides.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+      leftsum += arr[j];
+    }
+    for (let j = i + 1; j < arrLeng; j++) {
+      rightsum += arr[j];
+    }
+    if (leftsum == rightsum) {
+      return i;
+    }
+  }
+  if (leftsum != rightsum) {
```
Comment (prezlyataa) [inline tasks/45_equal-sides/equal-sides.js]: `!==`

File: tasks/42_first-n-smallest/first-n-smallest.js (hunk 1 above)
Comment (SofiMat) [reply tasks/42_first-n-smallest/first-n-smallest.js]: updated

Comment (SofiMat) [reply tasks/42_first-n-smallest/first-n-smallest.js]: updated

File: tasks/41_sort-by-bits/sort-by-bits.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+export function sortByBits(arr) {
+  arr.sort(function(a, b) {
```
[the remark this answers is not in the document]
Comment (SofiMat) [reply tasks/41_sort-by-bits/sort-by-bits.js]: updated

File: tasks/45_equal-sides/equal-sides.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  for (let i = 0; i < arrLeng; ++i) {
+    leftsum = 0;
+    rightsum = 0;
+    for (let j = 0; j < i; j++) {
+      leftsum += arr[j];
+    }
+    for (let j = i + 1; j < arrLeng; j++) {
+      rightsum += arr[j];
+    }
+    if (leftsum == rightsum) {
```
[the remark this answers is not in the document]
Comment (SofiMat) [reply tasks/45_equal-sides/equal-sides.js]: updated

File: tasks/45_equal-sides/equal-sides.js (hunk 2 above)
Comment (SofiMat) [reply tasks/45_equal-sides/equal-sides.js]: updated

**Pull request merged**